chef/cookbooks/neutron/recipes/common_agent.rb

```diff
@@ -257,10 +289,57 @@
         Chef::Log.info("#{name} usurped #{res[1].join(", ")} routes from #{bound_if}") unless res[1].empty?
       end
     end
+
+    execute "Neutron: add #{bound_if} to #{name}" do
+      command "ovs-vsctl add-port #{name} #{bound_if}"
+      not_if not_if_clause
+    end
     source = ::Nic.new(bound_if)
     # filter out cluster VIPs that a currently assigned to the interface
     addresses = source.addresses.select { |address| my_addresses.include? address }
     routes = source.routes
+
+    # Try to bring the cluster back up.
+    if handle_corosync_restart
+      service "start openais after network reconfiguration" do
+        service_name node[:corosync][:platform][:service_name]
+        action [:start]
+        not_if { addresses.empty? and routes.empty? }
```

Comment:
I'm not sure you should use the same `not_if_clause` because I'm unsure when it's evaluated :-) If it's during the compile phase, then it will work, but if it's in the other phase, then it will break.

And don't worry, corosync will get restarted on subsequent runs since the corosync cookbook should take care of that. Or am I missing something?